src/test/java/org/pih/petl/job/PetlScheduledExecutionTaskTest.java

```diff
@@ -0,0 +1,69 @@
+package org.pih.petl.job;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.pih.petl.ApplicationConfig;
+import org.pih.petl.SpringRunnerTest;
+import org.pih.petl.api.EtlService;
+import org.pih.petl.api.JobExecutionRepository;
+import org.pih.petl.job.schedule.PetlScheduledExecutionTask;
+import org.quartz.JobExecutionContext;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.util.Date;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class PetlScheduledExecutionTaskTest {
+
+    @Autowired
+    EtlService etlService;
+
+    @Autowired
+    PetlScheduledExecutionTask petlScheduledExecutionTask;
+
+    @Autowired
+    JobExecutionRepository jobExecutionRepository;
+
+    JobExecutionContext jobExecutionContext;
+
+    static {
+        SpringRunnerTest.setupEnvironment();
+    }
+
+    @Before
+    public void setup() {
+        jobExecutionContext = mock(JobExecutionContext.class);
+        when(jobExecutionContext.getFireTime()).thenReturn(new Date());
+        // override job test directory to only run jobs in test directory for this test
+        System.setProperty(ApplicationConfig.PETL_JOB_DIR, "src/test/resources/configuration/jobs/scheduler");
+    }
+
+    @Test
+    public void testRunningScheduledExecutionTaskShouldOnlyRunJobWithSchedule() throws Exception {
+        jobExecutionRepository.deleteAll();
```

Comment:
Shouldn't this go in an `@After` function?

Reply:
Yeah, there was reason I didn't put it there, but I think it was a "not sure why it doesn't work here" reason... :)

I'll try again, but may leave it as-is.

Reply:
Ah, I think I remember now... I think the other tests dirty the context... so, yeah, we should change all those tests to call this after as well.

I moved it to the "Before", which is at least a little better... double-checking that actually works...